### PR TITLE
new standard resolver option in ldc-download.in shell script

### LINKED ISSUE #1507: Use standard resolver in lxc-download
When filesystem eg. /dev/sda1 is mounted to **/srv/nginx** and **lnx.lnxpatch = /srv** is set too.

```
root@rastel:/srv# lxc-create -n nginx -t download

Setting up the GPG keyring
ERROR: Unable to fetch GPG key from keyserver.
lxc-create: lxccontainer.c: create_run_template: 1297 container creation template for nginx failed
lxc-create: utils.c: _recursive_rmdir: 170 _recursive_rmdir: **failed to delete /srv/nginx**
lxc-create: lxccontainer.c: container_destroy: 2411 Error destroying container directory for nginx
lxc-create: tools/lxc_create.c: main: 318 Error creating container nginx
```

Everything is OK, when
`root@rastel:/srv# lxc-create -n nginx -t gentoo`

### PR DESCRIPTION
**Pull request by AndrewElvisDeng**
Signed-off-by: Andrew Deng <adeng1433@gmail.com>
Fixes #1507

### CODE REVIEW

**Comment by stgraber** [comment]
You should also add a `--standard-resolver` CLI argument, that's pretty trivial to do by upting the getopt call and the loop after it as well as the help message.

**Review by brauner — CHANGES_REQUESTED** [review]
The commit message needs to be changed. "Implemented requested changes" doesn't tell us anything about what has been changed and why it has been changed. Right now I've no idea what this commit is supposed to be about. :)

**Comment by AndrewElvisDeng** [comment]
@brauner @stgraber We've updated the commit messages and added the `--standard-resolver` argument

File: templates/lxc-download.in [hunk 1]
```diff
@@ -126,10 +126,16 @@ gpg_setup() {
     return
   fi
 
+  
+
```
Comment (stgraber) [inline templates/lxc-download.in:130]: What's that about? :)

File: templates/lxc-download.in [hunk 2]
```diff
@@ -222,6 +228,7 @@ Optional arguments:
 [ --no-validate ]: Disable GPG validation (not recommended)
 [ --flush-cache ]: Flush the local copy (if present)
 [ --force-cache ]: Force the use of the local copy even if expired
+[ --standard-resolver <resolver> ]: Force the use of a specific standard resolver
```
Comment (stgraber) [inline templates/lxc-download.in:231]: This shouldn't take arguments, it should just be `--standard-resolver` and if passed should set DOWNLOAD_STANDARD_RESOLVER to `true`.

File: templates/lxc-download.in [hunk 3]
```diff
@@ -126,10 +126,16 @@ gpg_setup() {
...
   fi
 
+  
+
   echo "Setting up the GPG keyring"
 
   mkdir -p "${DOWNLOAD_TEMP}/gpg"
   chmod 700 "${DOWNLOAD_TEMP}/gpg"
+
+  if [ --standard-resolver "${STANDARD_RESOLVER:-}" ]; then
```
Comment (stgraber) [inline templates/lxc-download.in:136]: This is invalid shell, you should check for something like `"${DOWNLOAD_STANDARD_RESOLVER}" = "true"` and make sure to initialize `DOWNLOAD_STANDARD_RESOLVER` to `false` at the top of the file.

File: templates/lxc-download.in (hunk 1 above)
Comment (AndrewElvisDeng) [reply templates/lxc-download.in:130]: Don't know how that snuck in there...

File: templates/lxc-download.in [hunk 4]
```diff
@@ -130,6 +131,10 @@ gpg_setup() {
 
   mkdir -p "${DOWNLOAD_TEMP}/gpg"
   chmod 700 "${DOWNLOAD_TEMP}/gpg"
+
+  if [ --standard-resolver "${DOWNLOAD_STANDARD_RESOLVER}" = "true" ]; then
```
Comment (stgraber) [inline templates/lxc-download.in:135]: That should just be `if [ "${DOWNLOAD_STANDARD_RESOLVER}" = "true" ]; then`

File: templates/lxc-download.in [hunk 5]
```diff
@@ -130,6 +131,10 @@ gpg_setup() {
 
   mkdir -p "${DOWNLOAD_TEMP}/gpg"
   chmod 700 "${DOWNLOAD_TEMP}/gpg"
+  
+if [ "${DOWNLOAD_STANDARD_RESOLVER}" = "true" ]; then
```
Comment (stgraber) [inline templates/lxc-download.in:135]: Indent is broken somehow

File: templates/lxc-download.in [hunk 6]
```diff
@@ -265,6 +271,7 @@ while :; do
     --rootfs)      LXC_ROOTFS="$2"; shift 2;;
     --mapped-uid)  LXC_MAPPED_UID="$2"; shift 2;;
     --mapped-gid)  LXC_MAPPED_GID="$2"; shift 2;;
+    --standard-resolver) STANDARD_RESOLVER="$2"; shift 2;;
```
Comment (stgraber) [inline templates/lxc-download.in:274]: This should be:
```
--standard-resolver) STANDARD_RESOLVER="true"; shift 1;;
```

And I'd also move it up with the other flags, just under `--force-cache`

**Comment by stgraber** [comment]
Looks good, can you rebase this into just a single commit?

**Comment by AndrewElvisDeng** [comment]
@stgraber Done!

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**